src/ngx_http_lua_socket_tcp.c

```diff
@@ -4886,7 +5387,6 @@ ngx_http_lua_get_keepalive_peer(ngx_http_request_t *r, lua_State *L,
             cln = ngx_http_lua_cleanup_add(r, 0);
             if (cln == NULL) {
                 u->ft_type |= NGX_HTTP_LUA_SOCKET_FT_ERROR;
-                lua_settop(L, top);
```

Comment:
Why remove this? This is a security measure. Just to make sure the Lua stack is always balanced for this C function call.

Reply:
@agentzh 
There is no need to call `lua_settop`, since we don't touch Lua VM in `ngx_http_lua_get_keepalive_peer` now. We just use `u->socket_pool` directly.